test/task.h

```diff
@@ -229,4 +231,61 @@ UNUSED static int can_ipv6(void) {
   "Cygwin runtime hangs on listen+connect in same process."
 #endif
 
+#if !defined(MAP_ANONYMOUS)
+#if defined(MAP_ANON)
+#define MAP_ANONYMOUS MAP_ANON
+#else
+#define MAP_ANONYMOUS 0 // MAP_FILE
+#endif
+#endif
+
+/* Allocate a large, mostly-zero'd, mostly-readonly memory region */
+UNUSED static void *zalloc_2gb(void) {
+#if defined(_WIN32)
+  /* Win32 just can't allocate large blocks of memory reliably,
+   * and Win64 can't use them in IO,
+   * so we actually aren't going to test for it's behavior */
+  const size_t nbytes = (1u << 30u) + 1;
+  return calloc(1, nbytes);
+#else
+  const size_t nbytes = (1u << 31u) + 1;
+  char* const addr = (char*)0x20000000L + ((sizeof(void*) > 4) ? 0x200000000L : 0);
+  char* const endaddr = addr + nbytes;
+  char *p;
+  long pagesize;
+  int fd;
+  if (sizeof(void*) > 4) {
+    /* If we can, first just try to allocate something normally. */
+    p = calloc(1, nbytes);
+    if (p)
+      return p;
+  }
+  /* If this allocation would occupy most of memory, try instead just to make a
+   * large consecutive region of memory readable (ignoring its contents). */
```

Comment:
...that doesn't seem like a good idea at all.

Reply:
The alternative is that many of the buildbots (especially the 32-bit ones, but also it seems some 64-bit ones) can't run this test at all. And it seems to work fairly well on x86.